src/core.ts

```diff
@@ -147,18 +136,16 @@ export function isValidMerkleTree(tree: Bytes[]): boolean {
       if (l < tree.length) {
         return false;
       }
-    } else if (!equalsBytes(node, hashPair(tree[l]!, tree[r]!))) {
+    } else if (node !== hashPair(tree[l]!, tree[r]!)) {
```

Comment:
This is not right. `BytesLike` can be an array type, for which `===` is not the equality that we want. Additionally, for hex strings there are potential issues with inconsistent use of upper and lower case letters. We should keep an `equalsBytes` function, and for strings make sure to lowercase the string before comparing.

Reply:
I updated the code to use compare here. Does that feel ok, of should we be worried of performance issues. I'd rather not have multiple utils doing the same basic thing.

Reply:
No concerns about performance. I would prefer `const equalsBytes = (a, b) => compare(a, b) === 0` but up to you.

Reply:
AFAIK, there is only one location where such comparaison is needed. I would avoid adding a definition, and export, and an import for something that is only going to be used once.

Reply:
I also like `equalsBytes` but I agree with @Amxx I'd rather not reexport it now that it's not used too frequently.